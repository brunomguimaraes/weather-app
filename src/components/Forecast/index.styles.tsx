import styled from 'styled-components';

export const Container = styled.div`
  padding: 20px;
  background-color: ${props => props.theme.colors.secondary};
  border-radius: 5px;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
`;

export const ForecastDetails = styled.div`
  margin-top: 20px;
`;

export const ForecastTitle = styled.h4`
  font-size: 24px;
  margin-bottom: 20px;
  color: ${props => props.theme.colors.textPrimary};
`;

export const ForecastItemTitle = styled.h5`
  color: ${props => props.theme.colors.textSecondary};
`;

export const ForecastList = styled.ul`
  list-style-type: none;
  padding: 0;
`;

export const ForecastItem = styled.li`
  display: flex; // New
  justify-content: space-between; // New
  margin-bottom: 15px;
  padding: 10px;
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
`;

export const TimeOfDayForecast = styled.div`
  width: 48%; // This makes sure there's a little gap between the two child elements.
  display: flex;
  flex-direction: column;
  align-items: center; // This centers the items horizontally.
`;

export const Image = styled.img`
  width: 50px;
  height: 50px;
  margin-bottom: 10px;
`;

export const ForecastText = styled.p`
  margin-bottom: 5px;
  font-size: 16px;
  color: ${props => props.theme.colors.textSecondary};
`;
