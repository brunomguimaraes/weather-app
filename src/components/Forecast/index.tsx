import { WeatherForecast } from "../../services/api/weatherService";

import * as Styled from "./index.styles";

interface ForecastProps {
  forecastData: WeatherForecast;
}

const Forecast: React.FC<ForecastProps> = ({ forecastData }) => {
  const { periods } = forecastData.properties;

  return (
    <Styled.Container>
      <Styled.ForecastDetails>
        <Styled.ForecastTitle>7-Day Forecast</Styled.ForecastTitle>
        <Styled.ForecastList>
          {Array.from({ length: 7 }).map((_, dayIndex) => {
            const morningForecast = periods[dayIndex * 2];
            const nightForecast = periods[dayIndex * 2 + 1];

            return (
              <Styled.ForecastItem key={morningForecast.number}>
                <Styled.TimeOfDayForecast>
                  <Styled.ForecastItemTitle>
                    {morningForecast.name}
                  </Styled.ForecastItemTitle>
                  <Styled.Image
                    src={morningForecast.icon}
                    alt={`${morningForecast.name} Morning`}
                  />
                  <Styled.ForecastText>
                    <strong>Temperature:</strong> {morningForecast.temperature}{" "}
                    {morningForecast.temperatureUnit}
                  </Styled.ForecastText>
                  <Styled.ForecastText>
                    <strong>Wind:</strong> {morningForecast.windSpeed} from the{" "}
                    {morningForecast.windDirection}
                  </Styled.ForecastText>
                  <Styled.ForecastText>
                    <strong>Forecast:</strong>{" "}
                    {morningForecast.detailedForecast}
                  </Styled.ForecastText>
                </Styled.TimeOfDayForecast>

                <Styled.TimeOfDayForecast>
                  <Styled.ForecastItemTitle>
                    {nightForecast.name}
                  </Styled.ForecastItemTitle>
                  <Styled.Image
                    src={nightForecast.icon}
                    alt={`${nightForecast.name} Night`}
                  />
                  <Styled.ForecastText>
                    <strong>Temperature:</strong> {nightForecast.temperature}{" "}
                    {nightForecast.temperatureUnit}
                  </Styled.ForecastText>
                  <Styled.ForecastText>
                    <strong>Wind:</strong> {nightForecast.windSpeed} from the{" "}
                    {nightForecast.windDirection}
                  </Styled.ForecastText>
                  <Styled.ForecastText>
                    <strong>Forecast:</strong> {nightForecast.detailedForecast}
                  </Styled.ForecastText>
                </Styled.TimeOfDayForecast>
              </Styled.ForecastItem>
            );
          })}
        </Styled.ForecastList>
      </Styled.ForecastDetails>
    </Styled.Container>
  );
};

export default Forecast;
