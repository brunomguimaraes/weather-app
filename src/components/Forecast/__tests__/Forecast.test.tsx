import React from 'react';
import { render } from '@testing-library/react';
import Forecast from '..';

describe('<Forecast />', () => {
  const mockForecastData = {
    properties: {
      periods: [
        {
          number: 1,
          name: "Day 1",
          icon: "https://sample-icon-1.com",
          temperature: 20,
          temperatureUnit: "C",
          windSpeed: "10 mph",
          windDirection: "N",
          detailedForecast: "Sunny with a slight chance of rain."
        },
      ]
    }
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<Forecast forecastData={mockForecastData as any} />); // Casting as `any` for simplicity
    expect(asFragment()).toMatchSnapshot();
  });
});
