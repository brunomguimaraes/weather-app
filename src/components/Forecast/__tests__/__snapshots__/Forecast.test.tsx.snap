// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP bBiCsn"
  >
    <div
      class="sc-grXZZQ frLpQo"
    >
      <h4
        class="sc-dkjKgF bVYCTf"
      >
        7-Day Forecast
      </h4>
      <ul
        class="sc-hLclGa jvDsxC"
      >
        <li
          class="sc-eDnVMP cJxIJa"
        >
          <h5>
            Day 1
          </h5>
          <img
            alt="Day 1"
            class="sc-jSwlEQ kllzEk"
            src="https://sample-icon-1.com"
          />
          <p
            class="sc-gKHVLF fZILRc"
          >
            <strong>
              Temperature:
            </strong>
             20 C
          </p>
          <p
            class="sc-gKHVLF fZILRc"
          >
            <strong>
              Wind:
            </strong>
             10 mph from the N
          </p>
          <p
            class="sc-gKHVLF fZILRc"
          >
            <strong>
              Forecast:
            </strong>
             Sunny with a slight chance of rain.
          </p>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
